src/commands/createFunction/FunctionListStep.ts

```diff
@@ -49,8 +49,8 @@ export class FunctionListStep extends AzureWizardPromptStep<IFunctionWizardConte
                 if (options.templateId) {
                     const actualId: string = t.id.toLowerCase();
                     const expectedId: string = options.templateId.toLowerCase();
-                    // Check for a match with or without the language in the id
-                    return actualId === expectedId || actualId === `${expectedId}-${language.toLowerCase()}`;
+                    // check for only whole word matches and escape the "\b" since it's in a string, not a RegExp expression
+                    return new RegExp(`\\b${expectedId}\\b`, 'gmi').test(actualId);
```

Comment:
Just to be safe, you should use the "escape-string-regexp" npm package to escape expectedId. Same as https://github.com/microsoft/vscode-azuretools/blob/5999c2ad4423e86f22d2c648027242d8816a50e4/ui/src/callWithMaskHandling.ts#L24

Reply:
Trying to use escape, but it's causing problems with escaping `\b` characters.

![image](https://user-images.githubusercontent.com/5290572/92639904-55b00480-f291-11ea-81cd-a1a3b0b220ad.png)
![image](https://user-images.githubusercontent.com/5290572/92639849-429d3480-f291-11ea-8ec2-8113bd1c519d.png)

Reply:
Well you only have to escape `expectedId`, not your \b characters :P

Reply:
LOL...right 😔